core/network/src/protocol/sync.rs

```diff
@@ -441,6 +441,53 @@ impl<B: BlockT> ChainSync<B> {
 		})
 	}
 
+	/// Request syncing for the given block from given set of peers.
+	// This is similar to on_block_announce with unknown parent hash.
+	pub fn sync_fork(&mut self, peers: Vec<PeerId>, hash: &B::Hash, number: NumberFor<B>) ->
+		Vec<(PeerId, BlockRequest<B>)>
+	{
+		debug!(target: "sync", "Explicit sync request for block {:?} with {:?}", hash, peers);
+		if self.is_known(&hash) || self.is_already_downloading(&hash) {
```

Comment:
I'd say we need something a bit more explicit. The fork-sync requests should be kept around _forever_ until removed or the block is actually known, and the `is_already_downloading` check should be done periodically. We can imagine a case where `is_already_downloading` is true at the moment, but then for some reason the node stops downloading that block. Then our requested block still wouldn't be synced.